tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs

```diff
@@ -0,0 +1,105 @@
+using EStimLibrary.Core.SpatialModel;
+using Xunit;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace EStimLibrary.UnitTests.Core.SpatialModel
+{
+    /// <summary>
+    /// Test class for the LocalizationData record.
+    /// </summary>
+    public class LocalizationDataTests
+    {
+        /// <summary>
+        /// Test the basic construction of LocalizationData and 
+        /// ensure immutability (encapsulation).
+        /// </summary>
+        [Fact]
+        public void LocalizationData_Constructor()
+        {
+            // Arrange
+            var areasFullyContaining = new List<int> { 1, 2, 3 };
```

Comment:
Make sure to test different values for each method; try to think of edge cases! You can give your test methods the [Theory] flag instead of the [Fact] flag so you can use [InlineData(...)] or [MemberData(...)] flags to pass in multiple test values without needing to copy the body of the test method over and over again